src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -233,17 +251,24 @@ void YARPRobotStatePublisherModule::onRead(yarp::rosmsg::sensor_msgs::JointState
     iDynTree::JointIndex jntIndex;
     for (size_t i=0; i < v.name.size(); i++)
     {
-        jntIndex = model.getJointIndex(v.name[i]);
-
-        if (!(model.getJoint(jntIndex)->getNrOfDOFs()) || jntIndex == iDynTree::JOINT_INVALID_INDEX)
+        for (size_t j=0; j < model.getNrOfDOFs(); j++)
+        {
+            if (v.name[i] == model.getJointName(j))
+            {
+                jntIndex = model.getJointIndex(v.name[i]);
+                break;
+            }
+        }
```

Comment:
What is the purpose of this change?

Reply:
My bad, I didn't resolve the conflicts correctly after merging #548 to devel!  
Updated in https://github.com/robotology/idyntree/pull/533/commits/fd6f66f37697ba5ef6d524b825afb076c6cfd4e5